File: Java/Mahasiswa.java
package LATIHAN1DPBO2023.Java;

public class Mahasiswa {

    // atribut privat
    private String name;
    private int nim;
    private String studi;
    private String fakultas;

    // construtor
    public Mahasiswa(){
        this.name = "";
        this.nim = 0;
        this.studi = "";
        this.fakultas = "";
    }
    
    public Mahasiswa(String name, int nim, String studi, String fakultas)
    {
        this.name = name;
        this.nim = nim;
        this.studi = studi;
        this.fakultas = fakultas;
    }

    //getter
    public String getName()
    {
        return name;
    }
    public int getNim()
    {
        return nim;
    }
    public String getStudi()
    {
        return studi;
    }
    public String getFakultas()
    {
        return fakultas;
    }

    //setter
    public void setName(String name)
    {
        this.name = name;
    }
    public void setNim(int nim) {
        this.nim = nim;
    }
    public void setStudi(String studi)
    {
        this.studi=studi;
    }
    public void setFakultas(String fakultas) {
        this.fakultas = fakultas;
    }
}

File: C++/Mahasiswa.cpp
#include <iostream>
#include <string>

using namespace std;

class Mahasiswa {

    // atribut privat
    private:
        string name;
        int nim;
        string studi;
        string fakultas;
    
    // constructor
    public:
        Mahasiswa(){
            this->name = "";
            this->nim = 0;
            this->studi = "";
            this->fakultas = "";
        }

        Mahasiswa(string name, int nim, string studi, string fakultas){
            this->name = name;
            this->nim = nim;
            this->studi = studi;
            this->fakultas = fakultas;
        }

    // getter
    string getName()
    {
        return this->name;
    }
    int getNim()
    {
        return this->nim;
    }
    string getStudi()
    {
        return this->studi;
    }
    string getFakultas()
    {
        return this->fakultas;
    }

    // setter
    void setName(string name)
    {
        this->name = name;
    }
    void setNim(int nim)
    {
        this->nim = nim;
    }
    void setStudi(string studi)
    {
        this->studi = studi;
    }
    void setFakultas(string fakultas)
    {
        this->fakultas = fakultas;
    }
};



